sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonHighlighter.java

```diff
@@ -0,0 +1,137 @@
+import org.sonar.python.api.PythonKeyword;
+import org.sonar.python.api.PythonTokenType;
+import org.sonar.squidbridge.SquidAstVisitor;
+
+/**
+ * Colorizes Python code. Currently colorizes:
+ * <ul>
+ *   <li>
+ *     String literals. Example:
+ *     <pre>
+ *       "hello"
+ *       'hello'
+ *     </pre>
+ *   </li>
+ *   <li>
+ *     Keywords. Example:
+ *     <pre>
+ *       def
+ *     </pre>
+ *   </li>
+ *   <li>
+ *     Comments. Example:
+ *     <pre>
+ *        # some comment
+ *     </pre>
+ *   </li>
+ *   <li>
+ *     Doc strings. Examples:
+ *     <pre>
+ *        """ a doc string"""
+ *        ''' another doc string '''
+ *     </pre>
+ *   </li>
+ * </ul>
+ * Note that doc strings, such as:
+ * <pre>
+ *   """ a doc string"""
+ *   ''' another doc string '''
+ * </pre>
+ * are handled (i.e., colorized) like normal strings.
+ */
+public class PythonHighlighter extends SquidAstVisitor<Grammar> implements AstAndTokenVisitor {
+
+  private NewHighlighting newHighlighting;
+
+  private final SensorContext context;
+
+  public PythonHighlighter(SensorContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public void visitFile(@Nullable AstNode astNode) {
+    newHighlighting = context.newHighlighting();
+    InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates().is(getContext().getFile().getAbsoluteFile()));
+    newHighlighting.onFile(inputFile);
+  }
+
+  @Override
+  public void visitToken(Token token) {
+    if (token.getType().equals(PythonTokenType.STRING)) {
+      // case: string literal, including doc string
+      highlight(token, TypeOfText.STRING);
+
+    } else if (token.getType() instanceof PythonKeyword) {
+      // case: keyword
+      highlight(token, TypeOfText.KEYWORD);
+    }
+
+    for (Trivia trivia : token.getTrivia()) {
+      // case: comment
+      highlight(trivia.getToken(), TypeOfText.COMMENT);
+    }
+  }
+
+  @Override
+  public void leaveFile(@Nullable AstNode astNode) {
+    newHighlighting.save();
+  }
+
+  private void highlight(Token token, TypeOfText typeOfText) {
+    TokenLocation tokenLocation = new TokenLocation(token);
+    newHighlighting.highlight(tokenLocation.startLine(), tokenLocation.startLineOffset(), tokenLocation.endLine(), tokenLocation.endLineOffset(), typeOfText);
+
+    // if (typeOfText == TypeOfText.STRING) {
+    // String[] tokenLines = token.getValue().split("\n");
+    // int line = 0;
+    // for (int i = 0; i < tokenLines.length; i++) {
+    // int firstColumn;
+    // if (i == 0) {
+    // line = token.getLine();
+    // firstColumn = token.getColumn();
+    // } else {
+    // line++;
+    // firstColumn = 0;
+    // }
+    // newHighlighting.highlight(line, firstColumn, line, firstColumn + tokenLines[i].length(), typeOfText);
+    // }
+    // } else {
+    // newHighlighting.highlight(token.getLine(), token.getColumn(), token.getLine(), token.getColumn() + token.getValue().length(),
+    // typeOfText);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)